api/src/backend/api/v1/views.py

```diff
@@ -2565,3 +2572,130 @@ def get_serializer_context(self):
         context = super().get_serializer_context()
         context["allowed_providers"] = self.allowed_providers
         return context
+
+
+class LighthouseConfigViewSet(BaseRLSViewSet):
+    """
+    API endpoint for managing OpenAI API configuration.
+    """
+
+    filterset_fields = {
+        "name": ["exact", "icontains"],
+        "model": ["exact", "icontains"],
+        "is_active": ["exact"],
+        "inserted_at": ["gte", "lte"],
+    }
+    ordering_fields = ["name", "inserted_at", "updated_at", "is_active"]
+    ordering = ["-inserted_at"]
+
+    def get_queryset(self):
+        return LighthouseConfig.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "create":
+            return LighthouseConfigCreateSerializer
+        elif self.action in ["update", "partial_update"]:
+            return LighthouseConfigUpdateSerializer
+        return LighthouseConfigSerializer
+
+    @action(detail=True, methods=["get"])
+    def check_connection(self, request, pk=None):
+        """
+        Check the connection to the OpenAI API.
+        """
+        instance = self.get_object()
+        if not instance.api_key_decoded:
+            return Response(
+                {"detail": "API key is invalid or missing."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            client = openai.OpenAI(
+                api_key=instance.api_key_decoded,
+            )
+            models = client.models.list()
+            return Response(
+                {
+                    "detail": "Connection successful!",
+                    "available_models": [model.id for model in models.data],
+                },
+                status=status.HTTP_200_OK,
+            )
+        except Exception as e:
+            return Response(
+                {"detail": f"Connection failed: {str(e)}"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+    def create(self, request, *args, **kwargs):
+        """Create new AI configuration"""
+        if LighthouseConfig.objects.filter(tenant_id=request.tenant_id).exists():
+            return Response(
+                {"detail": "AI configuration already exists. Use PUT to update."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
```

Comment:
We can use the serializers validate method like in `provider-groups` and this way specify which attribute produce the error

Reply:
https://www.django-rest-framework.org/api-guide/validators/#uniquetogethervalidator

Check `ProcessorCreateSerializer` for reference.